commit ce312c83543787bdd921ed1b429f041f74648d77
Author: Lukas Wunner <lukas@wunner.de>
Date:   Wed Jan 16 12:22:32 2019 +0100

    bcm2835-mmc: Fix DMA channel leak
    
    The BCM2835 MMC host driver requests a DMA channel on probe but neglects
    to release the channel in the probe error path and on driver unbind.
    
    I'm seeing this happen on every boot of the Compute Module 3: On first
    driver probe, DMA channel 2 is allocated and then leaked with a "could
    not get clk, deferring probe" message. On second driver probe, channel 4
    is allocated.
    
    Fix it.
    
    Signed-off-by: Lukas Wunner <lukas@wunner.de>
    Cc: Frank Pavlic <f.pavlic@kunbus.de>

diff --git a/drivers/mmc/host/bcm2835-mmc.c b/drivers/mmc/host/bcm2835-mmc.c
index 905075c94232..fd2225a388c4 100644
--- a/drivers/mmc/host/bcm2835-mmc.c
+++ b/drivers/mmc/host/bcm2835-mmc.c
@@ -1503,6 +1503,8 @@ static int bcm2835_mmc_probe(struct platform_device *pdev)
 
 	return 0;
 err:
+	if (host->dma_chan_rxtx)
+		dma_release_channel(host->dma_chan_rxtx);
 	mmc_free_host(mmc);
 
 	return ret;
@@ -1548,6 +1550,9 @@ static int bcm2835_mmc_remove(struct platform_device *pdev)
 
 	tasklet_kill(&host->finish_tasklet);
 
+	if (host->dma_chan_rxtx)
+		dma_release_channel(host->dma_chan_rxtx);
+
 	mmc_free_host(host->mmc);
 	platform_set_drvdata(pdev, NULL);
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1387_ce312c83543787bdd921ed1b429f041f74648d77.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1387_ce312c83543787bdd921ed1b429f041f74648d77.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
